### ISSUE TITLE
Rtl local

### ISSUE THREAD

**Issue by ashrafalzyoud** [sk-ys/redmine_enhanced_ux#19]
Can u support the rtl languages 
Because the css all in ltr 
I need a lot of changed

**Comment by ashrafalzyoud** [comment]
![rtl](https://github.com/user-attachments/assets/4111d6e9-5983-4eba-9105-c0e45dde951b)
this pic describe

**Comment by ashrafalzyoud** [comment]
realy thx thx for your plugin for me its perfect perfect
im try this code to support rtl
@media screen and (min-width: 900px) {
  #content {
    // Styles for LTR
    [dir="ltr"] & {
      > div.contextual {
        position: absolute;
        right: 15px; // Align to the right for LTR
      }

      > p.query-totals {
        margin: 0;
      }
    }

    // Styles for RTL
    [dir="rtl"] & {
      > div.contextual {
        position: absolute;
        left: 15px; // Align to the left for RTL
      }

      > p.query-totals {
        margin: 0;
      }
    }
  }
but its not working

**Comment by ashrafalzyoud** [comment]
i found the solution but plz check it to suppour RTL language
```
const currentLang = document.documentElement.lang || 'en';
const isRtl = currentLang === 'ar'; // Check for RTL languages
if (isRtl) {
  document.body.classList.add('rtl');
}
```


```
.rtl #content > div.contextual {
  position: absolute;
  left: 15px;  /* Align to the left for RTL */
  right: auto; /* Ensure no right alignment */
}
```
becuase the problem in div.contextual 
@sk-ys

**Comment by sk-ys** [comment]
@ashrafalzyoud 
I understand your objective, but I expect it to be difficult.
There are many changes that need to be made.

**Comment by ashrafalzyoud** [comment]
look to this pic 
i made just only in div content
its work as i want
![enhanced](https://github.com/user-attachments/assets/27cce7aa-2261-4583-ba6a-88f3b58f2776)
```
.rtl #content > div.contextual {
  position: absolute;
  left: 15px;  /* Align to the left for RTL */
  right: auto; /* Ensure no right alignment */
}
```

**Comment by sk-ys** [comment]
@ashrafalzyoud
Hi, I have created a new branch and am currently testing it. I tried modifying the Issue list view.
Please check the following:
https://github.com/sk-ys/redmine_enhanced_ux/tree/issue-19-support-rtl

**Comment by ashrafalzyoud** [comment]
thx thx thx for your help
if u can added 
this default ltr
```
@media (min-width: 900px) {
    #flash_notice {
        position: fixed;
        max-width: calc(100vw - 40px);
        left: 20px;
        box-sizing: border-box;
        z-index: 300;
    }
```
in rtl
```
@media (min-width: 900px) {
    #flash_notice {
        max-width: calc(100vw - 40px);
        right: 20px;
        box-sizing: border-box;
        z-index: 300;
    }
```

**Comment by sk-ys** [comment]
@ashrafalzyoud
Hi, I have fixed some designs in RTL view. Please check the following.
As a known issue, support for RTL view in the Gantt chart is not yet available.
It seems necessary to first ensure that the Gantt chart in Redmine itself displays correctly in RTL.
https://github.com/sk-ys/redmine_enhanced_ux/tree/issue-19-support-rtl

**Comment by ashrafalzyoud** [comment]
Very thx for u
I will try it as soon as
We care more about issues 

But plz care about dependents function 
And when add child

**Comment by sk-ys** [comment]
@ashrafalzyoud
I addressed the issue of adding a child issue in #36.
This update is committed to the dev branch (which does not support RTL).
https://github.com/sk-ys/redmine_enhanced_ux/tree/dev

**Comment by ashrafalzyoud** [comment]
When u added support rtl with dev

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sk-ys** [comment]
@ashrafalzyoud
If you confirm these changes in the `issue-19-support-rtl` branch, I will merge it into the dev branch.
I will make progress on PR #37.

**Comment by ashrafalzyoud** [comment]
every things seem very perfect just some note
1- @media (min-width: 900px) {
    #flash_notice:not(.fadeout_notice):not(.displayed) {
      animation-name: fade_in_notice;
      animation-duration: 2s;
      animation-fill-mode: forwards;
    }
    @keyframes fade_in_notice {
      0% {
        opacity: 0;
      }
      100% {
        opacity: 90%;
      }
    }
    #flash_notice {
      position: static;  // im change it because the flash merge in subject
      max-width: calc(100vw - 40px);
      left: 20px;
      box-sizing: border-box;
      z-index: 300;
    }
  }
![image](https://github.com/user-attachments/assets/3ad6ef5f-c2ca-48bb-955e-af3a4aa25acf)


2- if u can let the flash notice not disapper after secounds or tell me how i can let it without timer
3- its very important
when im creat new child the (id number not appear inside the screen child or in parent child like deafult redmine

**Comment by sk-ys** [comment]
1. I prefer not to set it to "static" because it causes the content to shift downward. If it’s in the way, you can hide it using the close button. Also, in the latest dev branch, the notice bar is now draggable.


2. The notice bar will not disappear while the mouse cursor is hovering over it. Currently, there is no setting to keep it visible indefinitely.


3. The notice bar does not display when clicking "create" or "create and follow." I apologize for this limitation, and support for this is not planned at the moment.

**Comment by ashrafalzyoud** [comment]
thx for your answers 
3- in default redmine when u added subtask (create) the notice bar with id its appear, why we need it because the users have many subtask added he need know the id which added

**can u now merge rtl wirh dev to download it and test every things plzz**

**Comment by ashrafalzyoud** [comment]
u can close this issue

**Comment by sk-ys** [comment]
@ashrafalzyoud 
I fixed the notice-bar-related issue in issue #41 and merged it into the Dev branch.
